extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -830,4 +884,559 @@ private Table readTableInternal(
     public Catalog catalog() {
         return catalog;
     }
+
+    /**
+     * Append the provided deephaven table as a new partition to the existing iceberg table in a single snapshot.
+     *
+     * @param tableIdentifier The identifier string for the iceberg table to append to
+     * @param dhTable The deephaven table to append
+     * @param instructions The instructions for customizations while writing, or null to use default instructions
+     */
+    public void append(
```

Comment:
We aren't currently providing a way to add partitioned data to an Iceberg table, but we should create a ticket for this functionality.

Reply:
Will check with Ryan/Devin if I should add this as part of this PR itself, or should start a separate ticket/PR.

Reply:
Do we support reading partitioned Iceberg?

Reply:
Yes, we do. cc: @lbooker42